llvm/test/TableGen/riscv-target-def.td

```diff
@@ -51,6 +51,14 @@ def Feature64Bit
 def FeatureDummy
     : SubtargetFeature<"dummy", "Dummy", "true", "Dummy">;
 
+class RISCVProfile<string name, list<SubtargetFeature> features>
+    : SubtargetFeature<name, "RISCVProfile", NAME,
+                       "RISC-V " # name # " profile", features>;
+
+def RVI20U32 : RISCVProfile<"rvi20u32", [Feature32Bit, FeatureStdExtI]>;
+def RVI20U64 : RISCVProfile<"rvi20u64", [Feature64Bit, FeatureStdExtI]>;
```

Comment:
It might be worth having a test with at least one extension implication (e.g. D, which should pull in F).

Reply:
We don't handle implications in TableGen, it is in `RISCVISAInfo` where we parse the march string. But yeah, I added `F` which implies `Zicsr`.